doc/experiment.rst

```diff
@@ -12,316 +38,231 @@ Jupyter notebook, interactive python session, or Python file and use the
-
-.. code-block:: Python
+Defining workflow stages requires the utilization of functions associated
+with the ``Experiment`` object. The Experiment object is intended to be instantiated
+once and utilized throughout the workflow runtime.
 
-  from smartsim import Experiment
+==========
+ Launchers
+==========
 
-  exp = Experiment("simple", launcher="local")
+The Experiment API *interfaces* both locally and with four
+different job schedulers that SmartSim refers to as `launchers`. `Launchers`
+provide SmartSim the ability to construct and execute complex workloads
+on HPC systems with schedulers (workload managers) like Slurm, or PBS.
 
-  settings = exp.create_run_settings("echo", exe_args="Hello World")
-  model = exp.create_model("hello_world", settings)
+SmartSim currently supports 5 `launchers`:
+  1. ``local``: for single-node, workstation, or laptop
+  2. ``slurm``: for systems using the Slurm scheduler
+  3. ``pbs``: for systems using the PBSpro scheduler
+  4. ``cobalt``: for systems using the Cobalt scheduler
+  5. ``lsf``: for systems using the LSF scheduler
+  6. ``auto``: have SmartSim auto-detect the launcher to use.
 
-  exp.start(model, block=True)
-  print(exp.get_status(model))
+If `launcher="auto"` is used, the experiment will attempt to find a launcher
+on the system, and use the first one it encounters. If a launcher cannot
+be found or no launcher parameter is provided, the default value of
+`launcher="local"` will be used.
 
-If the launcher has been specified, or auto-detected through setting
-``launcher=auto`` in the Experiment initialization, the ``create_run_settings``
-method will automatically create the appropriate ``RunSettings`` object and
-return it.
+A `launcher` enables SmartSim to interact with the compute system
+programmatically through the Experiment python user interface.
+SmartSim users do not have to leave the Jupyter Notebook,
+Python REPL, or Python script to launch, query, and interact with their jobs.
 
-For example with Slurm
+To specify a specific launcher, one argument needs to be provided
+to the ``Experiment`` initialization.
 
-.. code-block:: Python
-
-  from smartsim import Experiment
-
-  exp = Experiment("hello_world_exp", launcher="slurm")
-  srun = exp.create_run_settings(exe="echo", exe_args="Hello World!")
-
-  # helper methods for configuring run settings are available in
-  # each of the implementations of RunSettings
-  srun.set_nodes(1)
-  srun.set_tasks(32)
-
-  model = exp.create_model("hello_world", srun)
-  exp.start(model, block=True, summary=True)
-
-  print(exp.get_status(model))
-
-The above will run ``srun -n 32 -N 1 echo Hello World!``, monitor its
-execution, and inform the user when it is completed. This driver script can be
-executed in an interactive allocation, or placed into a batch script as follows:
+.. code-block:: python
 
-.. code-block:: bash
+    from smartsim import Experiment
+
+    exp = Experiment("name-of-experiment", launcher="local")  # local launcher
```

Comment:
I think we may want to just show one or two examples, and possibly in two separated code blocks, introduced by a sentence, like: "for example, to set up a local launcher, the ``Experiment`` would be specified as follows." and then maybe just again the explanation of ``auto`` or of a WLM. I think you explained thoroughly that ``launcher`` is the parameter one has to set.